### PR TITLE
Proposal PowerPC tests

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI https://trello.com/c/DRGWzT1s/504-2-storageng-make-proposal-work-with-power.

* Add tests to check proposal matches bootloader requirements for ppc.
* File `proposal_test` was too long, so it is splitted.
* Created files `proposal_scenarios_x86_test` and `proposal_scenarios_ppc_test`.

### CODE REVIEW

File: test/support/proposal_context.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    allow(Yast::Arch).to receive(:x86_64).and_return(architecture == :x86)
+    allow(storage_arch).to receive(:ppc_power_nv?).and_return(ppc_power_nv)
+    allow(Y2Storage::StorageManager.instance.storage).to receive(:arch).and_return(storage_arch)
+    allow(storage_arch).to receive(:efiboot?).and_return(false)
+    allow(storage_arch).to receive(:x86?).and_return(architecture == :x86)
+    allow(storage_arch).to receive(:ppc?).and_return(architecture == :ppc)
+    allow(storage_arch).to receive(:s390?).and_return(architecture == :s390)
+  end
+
+  subject(:proposal) { described_class.new(settings: settings) }
```
Comment (ancorgs) [inline test/support/proposal_context.rb]: I wouldn't place `subject` in a shared context

File: test/proposal_scenarios_ppc_test.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+require_relative "spec_helper"
+require "storage"
+require "y2storage"
+require_relative "support/proposal_context"
+
+describe Y2Storage::Proposal do
+  describe "#propose" do
+    using Y2Storage::Refinements::TestDevicegraph
+    using Y2Storage::Refinements::SizeCasts
+    using Y2Storage::Refinements::DevicegraphLists
```
Comment (ancorgs) [inline test/proposal_scenarios_ppc_test.rb]: I would say none of these three refinements are really used here

File: test/proposal_scenarios_ppc_test.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    let(:scenario) { "empty_hard_disk_50GiB" }
+
+    context "in a PPC64 bare metal (PowerNV)" do
+      let(:ppc_power_nv) { true }
+      let(:expected_scenario) { "ppc_power_nv" }
+
+      context "if requires a /boot partition" do
+        let(:lvm) { true }
+
+        it "creates a /boot partition" do
```
Comment (ancorgs) [inline test/proposal_scenarios_ppc_test.rb]: The description does not match the `expect`. This does not check specifically for the `/boot` partition. So I would use,
```ruby
context "if requires a /boot partition" do
  let(:lvm) { true }
  include_examples "proposed layout"
end
```
which would print "it proposes the expected layout". That's what the test actually does.

File: test/proposal_scenarios_ppc_test.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    end
+
+    context "in a PPC64 non-PowerNV" do
+      let(:ppc_power_nv) { false }
+      let(:expected_scenario) { "ppc_non_power_nv" }
+
+      context "if requires a PReP partition" do
+        let(:lvm) { false }
+
+        it "creates a PReP partition" do
```
Comment (ancorgs) [inline test/proposal_scenarios_ppc_test.rb]: Same than above

File: test/proposal_scenarios_ppc_test.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          proposal.propose
+          expect(proposal.devices.to_str).to eq expected.to_str
+        end
+      end
+
+      context "if requires /boot and PReP partitions" do
+        let(:lvm) { true }
+
+        it "creates /boot and PReP partitions" do
```
Comment (ancorgs) [inline test/proposal_scenarios_ppc_test.rb]: Same than above, again

File: test/proposal_scenarios_x86_test.rb [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+require "storage"
+require "y2storage"
+require_relative "support/proposal_examples"
+require_relative "support/proposal_context"
+
+describe Y2Storage::Proposal do
+  describe "#propose" do
+    using Y2Storage::Refinements::TestDevicegraph
+    using Y2Storage::Refinements::SizeCasts
+    using Y2Storage::Refinements::DevicegraphLists
```
Comment (ancorgs) [inline test/proposal_scenarios_x86_test.rb]: Same than above, these refinements don't look necessary at first sight.

File: test/proposal_test.rb [hunk 7]
```diff
@@ -24,156 +24,17 @@
 require "storage"
 require "y2storage"
 require_relative "support/proposal_examples"
```
Comment (ancorgs) [inline test/proposal_test.rb]: `proposal_examples` is not used here, I believe. Which makes me thing it should be called `proposal_scenario_examples`, BTW.

File: test/proposal_scenarios_x86_test.rb (hunk 6 above)
Comment (joseivanlopez) [inline test/proposal_scenarios_x86_test.rb]: You are right, they are not used here.

File: test/proposal_test.rb (hunk 7 above)
Comment (joseivanlopez) [inline test/proposal_test.rb]: Some tests here use it:

```
      context "if no disk is enforced for '/'" do
        let(:root_device) { nil }
        let(:yaml_suffix) { "sdb_root_device" }

        include_examples "proposed layout"
        ...
```
so, we have to include it.

File: test/proposal_scenarios_ppc_test.rb (hunk 2 above)
Comment (joseivanlopez) [inline test/proposal_scenarios_ppc_test.rb]: Really, only `using Y2Storage::Refinements::TestDevicegraph` was necessary, but we can remove it if shared proposal examples are used.

File: test/support/proposal_context.rb (hunk 1 above)
Comment (joseivanlopez) [inline test/support/proposal_context.rb]: Ok

File: test/proposal_scenarios_ppc_test.rb (hunk 3 above)
Comment (joseivanlopez) [inline test/proposal_scenarios_ppc_test.rb]: Ok

File: test/proposal_test.rb (hunk 7 above)
Comment (ancorgs) [inline test/proposal_test.rb]: Oh, I overlooked that one

**Pull request merged**